### PR TITLE
Olgo 7 implement jwt

### PR DESCRIPTION
**Pull request by 0xLeif**
Closes: #7

### CODE REVIEW

**Review by justindbeltran — APPROVED** [review]

**Review by justindbeltran — APPROVED** [review]

File: olGO/Extensions/JSONDecoder+olGO.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//
+//  Created by Zach Eriksen on 4/11/20.
+//  Copyright © 2020 oneleif. All rights reserved.
+//
+
+import Foundation
+
+extension JSONDecoder {
+    static var dateDecoder: JSONDecoder = {
+        let formatter = DateFormatter()
```
Comment (arkus7) [inline olGO/Extensions/JSONDecoder+olGO.swift:13]: Can't we just use `ISO8601DateFormatter` here?

```swift
let formatter = ISO8601DateFormatter()
formatter.formatOptions =  [.withInternetDateTime, .withFractionalSeconds]
```

File: olGO/Models/User.swift [hunk 2]
```diff
@@ -23,17 +24,19 @@ struct SocialInformation: Codable {
     var location: String = ""
 }
 
-struct User: Codable {
+final class User: SQLiteModel, Codable {
```
Comment (arkus7) [inline olGO/Models/User.swift]: What else do we want to store in memory sqlite? Is it worth it?

**Review by arkus7 — APPROVED** [review]

File: olGO/Models/User.swift (hunk 2 above)
Comment (0xLeif) [reply olGO/Models/User.swift]: I was debating if we should store the user... Only other things that might be nice to store are Posts...

Also I'd probably switch `memory` to a `file` to cache values on device and not use UserDefaults or CoreData...

File: olGO/API/API.swift [hunk 3]
```diff
@@ -76,41 +82,65 @@ class API {
...
+        
+        FLite.prepare(model: User.self)
+        
+        return API()
     }()
+    
     // Configuations
-    let path = "http://localhost:8080/api"
-//    let path: String = "https://olapi-develop.vapor.cloud/api"
+    //    let path = "http://localhost:8080/api"
```
Comment (ambid17) [inline olGO/API/API.swift]: build flag?

**Review by ambid17 — APPROVED** [review]

File: olGO/Extensions/JSONDecoder+olGO.swift (hunk 1 above)
Comment (0xLeif) [reply olGO/Extensions/JSONDecoder+olGO.swift:13]: Getting this error:
```
Cannot convert value of type 'ISO8601DateFormatter' to expected argument type 'DateFormatter'
```

**Pull request merged**